url-detector/src/main/java/com/linkedin/urls/detection/DomainNameReader.java

```diff
@@ -399,8 +410,11 @@ public ReaderNextState readDomainName() {
           _reader.goBack();
           done = true;
         } else {
+          if (isAllHexSoFar && !CharUtils.isHex(curr)) {
+            _numeric = false;
+          }
           //if its not numeric, remember that; excluded x/X for hex ip addresses.
-          if (curr != 'x' && curr != 'X' && !CharUtils.isNumeric(curr)) {
```

Comment:
hex ip addresses should start with ```0x```.
We should exclude skip ```x``` if it doesn't appear at the start after ```0```